cuenca/http/client.py

```diff
@@ -135,7 +138,9 @@ def request(
             **kwargs,
         )
         self._check_response(resp)
-        return resp.json()
+        if resp.headers['Content-Type'] == 'application/json':
```

Comment:
Is this necessary? It should be the default

Reply:
When the response is of type pdf, the body is not a dictionary so .json () would fail.

<img width="521" alt="Screen Shot 2020-12-21 at 14 35 31" src="https://user-images.githubusercontent.com/25915735/102819871-ea9ac180-4399-11eb-9d41-790df714e318.png">
<img width="425" alt="Screen Shot 2020-12-21 at 14 35 22" src="https://user-images.githubusercontent.com/25915735/102819868-e8d0fe00-4399-11eb-9269-4a4cac78a011.png">

Reply:
what about this?
```python
from json.decoder import JSONDecodeError

try:
    content = resp.json()
except JSONDecodeError:
    content = resp.content
return content
```

This solves a few issues:
1. the most common case is `resp.json()`, which means the `if` is unnecessary in nearly all cases
2. there's no reason to envelope `resp.content` in a `dict` (unless I'm missing something)
3. avoid having multiple `return` statements

Reply:
my recommendation would change the type hint for the return type of this `.request()` and all the methods that call it